samtranslator/parser/parser.py

```diff
@@ -55,4 +55,5 @@ def _validate(self, sam_template, parameter_values):
                     ]
                 )
 
-        SamTemplateValidator.validate(sam_template)
+        validator = SamTemplateValidator()
+        validation_errors = validator.validate(sam_template)
```

Comment:
Added a `LOG.warn()`.
This will also be displayed when calling `sam validate` as the parser is called by the [SamTemplateValidator from the CLI](https://github.com/aws/aws-sam-cli/blob/develop/samcli/commands/validate/lib/sam_template_validator.py#L68).

This is how it looks when adding an Api with an empty `StageName` to a template and calling `sam validate`:
```
>> samdev validate
2021-06-29 16:28:42 Loading policies from IAM...
2021-06-29 16:28:46 Finished loading policies from IAM.
2021-06-29 16:28:46 Template schema validation reported the following errors: [Resources/MyApi/Properties/StageName] Must not be empty
Template provided at '/path/to/template.yaml' was invalid SAM Template.
Error: [InvalidResourceException('MyApi', "Missing required property 'StageName'.")] ('MyApi', "Missing required property 'StageName'.")
```